html/BrokerNode.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+require_once("requests/IriWrapper.php");
+require_once("requests/IriData.php");
+require_once("requests/NodeMessenger.php");
+require_once("iota-php-port/PrepareTransfers.php");
+
+class BrokerNode
+{
+    public static $chunksToAttach = null;
+    public static $chunksToVerify = null;
+
+    public static $IriWrapper = null;
+    public static $NodeMessenger = null;
+
+    public static $iriRequestInProgress = false;
+
+
+    private static function initIri()
+    {
+        if (is_null(self::$IriWrapper)) {
+            self::$IriWrapper = new IriWrapper();
+        }
+    }
+
+    private static function initMessenger()
+    {
+        if (is_null(self::$NodeMessenger)) {
+            self::$NodeMessenger = new NodeMessenger();
+        }
+    }
+
+    public static function processNewChunk(&$chunk)
+    {
+        try {
+            if (self::dataNeedsAttaching($chunk)) {
+                self::getTransactionData($chunk);
+                self::sendToHookNode($chunk);
+            } else {
+                // move on to the next chunk
+                /*
+                 * TODO: is there anything specific we want to do if it's 
+                 * already attached?
+                 */
+            }
+        } catch
+        (Exception $e) {
+            echo "Caught exception: " . $e->getMessage();
+            // something went wrong during our check, do something about it
+        }
+    }
+
+    public static function dataNeedsAttaching($request)
+    {
+        $command = new stdClass();
+        $command->command = "findTransactions";
+        $command->addresses = array($request->address);
+
+        BrokerNode::$iriRequestInProgress = true;
+
+        self::initIri();
+        $result = self::$IriWrapper->makeRequest($command);
+
+        BrokerNode::$iriRequestInProgress = false;
+
+        if (!is_null($result) && property_exists($result, 'hashes')) {
+            return count($result->hashes) == 0;
+        } else {
+            throw new Exception('findTransactions failed!');
+        }
+    }
+
+    public static function getTransactionData(&$request)
+    {
+        $trytesToBroadcast = NULL;
+
+        $request->value = IriData::$txValue;
+        $request->tag = IriData::$oysterTag;
+
+        try {
```

Comment:
This can be simplified (same way as your earlier PR)

Reply:
I will change this.